src/form/Form.jsx

```diff
@@ -334,24 +334,16 @@ export default class Form extends React.Component {
     } = this.props
 
     return (
-      <form autoComplete="off" {...other} name={name} onSubmit={this.handleSubmit}>
-        {typeof children === 'function' ? children({
-          value: this.state.value,
-          pristine: Immutable.is(value, this.state.value),
-          submitting: this.state.submitting,
-          reset: () => this.reset()
-        }) : children}
-      </form>
+      <FormScopeValueContext.Provider value={{ instance: this, state: this.state }}>
```

Comment:
:point_up: 
It looks like passing just `value={this}` does not trigger the necessary rerender on the component. This is quite tricky, as in the legacy context API we were passing just `this`, and each call of `this.setState` did trigger a rerender.
With the new API this is not the case. The hacky workaround was to pass the state as well so that a rerender is forced.